Core/GDCore/Project/Variable.h

```diff
@@ -26,18 +27,58 @@ namespace gd {
  */
 class GD_CORE_API Variable {
  public:
+  enum Type {
+    // Primitive types
+    String,
+    Number,
+    Boolean,
+
+    // Structural types

```

Comment:
Let's not use the term "structural type" at all, it's for another thing in computer science.
Let's call then "Collection types".